materialize-sqlserver/sqlgen.go

```diff
@@ -40,8 +40,8 @@ var sqlServerDialect = func(collation string, defaultSchema string) sql.Dialect
 		stringType = "nvarchar"
 	}
 
-	var textType = fmt.Sprintf("%s(MAX) COLLATE %s", stringType, collation)
-	var textPKType = fmt.Sprintf("%s(900) COLLATE %s", stringType, collation)
+	var textType = fmt.Sprintf("%s(max) collate %s", stringType, strings.ToLower(collation))
```

Comment:
It seems like there is a need to do a case-insensitive match in `(MigrationSpecs).FindMigrationSpec`? That would make this more robust, rather than relying on the `targetDDLs` always having to be lowercased inputs.